src/NSwag.SwaggerGeneration.AspNetCore/AspNetCoreToSwaggerGenerator.cs

```diff
@@ -0,0 +1,257 @@
+                {
+                    Path = path,
+                    Method = GetSwaggerOperationMethod(apiDescription.HttpMethod),
+                    Operation = new SwaggerOperation
+                    {
+                        IsDeprecated = method.GetCustomAttribute<ObsoleteAttribute>() != null,
+                        OperationId = GetOperationId(document, controllerType.Name, method)
+                    }
+                };
+
+                operations.Add(new Tuple<SwaggerOperationDescription, ApiDescription, MethodInfo>(operationDescription, apiDescription, method));
+            }
+
+            await AddOperationDescriptionsToDocumentAsync(document, controllerType, operations, swaggerGenerator, schemaResolver).ConfigureAwait(false);
+        }
+
+        private async Task AddOperationDescriptionsToDocumentAsync(SwaggerDocument document, Type controllerType, List<Tuple<SwaggerOperationDescription, ApiDescription, MethodInfo>> operations, SwaggerGenerator swaggerGenerator, SwaggerSchemaResolver schemaResolver)
+        {
+            var allOperation = operations.Select(t => t.Item1).ToList();
+            foreach (var tuple in operations)
+            {
+                var operation = tuple.Item1;
+                var apiDescription = tuple.Item2;
+                var method = tuple.Item3;
+
+                var addOperation = await RunOperationProcessorsAsync(document, apiDescription, controllerType, method, operation, allOperation, swaggerGenerator, schemaResolver).ConfigureAwait(false);
+                if (addOperation)
+                {
+                    var path = operation.Path.Replace("//", "/");
+
+                    if (!document.Paths.ContainsKey(path))
+                        document.Paths[path] = new SwaggerOperations();
+
+                    if (document.Paths[path].ContainsKey(operation.Method))
+                    {
+                        throw new InvalidOperationException("The method '" + operation.Method + "' on path '" + path + "' is registered multiple times " +
+                            "(check the DefaultUrlTemplate setting [default for Web API: 'api/{controller}/{id}'; for MVC projects: '{controller}/{action}/{id?}']).");
+                    }
+
+                    document.Paths[path][operation.Method] = operation.Operation;
+                }
+            }
+        }
+
+        private async Task<SwaggerDocument> CreateDocumentAsync(AspNetCoreToSwaggerGeneratorSettings settings)
+        {
+            var document = !string.IsNullOrEmpty(settings.DocumentTemplate) ? await SwaggerDocument.FromJsonAsync(settings.DocumentTemplate).ConfigureAwait(false) : new SwaggerDocument();
+
+            document.Generator = "NSwag v" + SwaggerDocument.ToolchainVersion + " (NJsonSchema v" + JsonSchema4.ToolchainVersion + ")";
+            document.Consumes = new List<string> { "application/json" };
+            document.Produces = new List<string> { "application/json" };
```

Comment:
I think this is fine for a first cut, but we need to take a second look at this.